pkg/parser/ast.go

```diff
@@ -12,8 +12,13 @@ type Node interface {
 	Type() *Type
 }
 
+type Returner interface {
+	AlwaysReturns() bool
+}
+
 type Program struct {
-	Statements []Node
+	Statements       []Node
+	guaranteedReturn *lexer.Token
```

Comment:
This field is leftover from the previous implementation.

Reply:
oh, looks like the linter beat me to it

Reply:
oups, sorry. fixed.

Reply:
I should turn up the linter strictness.
Will do in follow up pr